### ISSUE TITLE
Fields merging not works as expected for Hocon.Extensions.Configuration

### ISSUE THREAD

**Issue by iron9light** [akkadotnet/HOCON#89]
Input:
```
a {
  aa: 1
  bb: "2"
}

a {
  cc: 3.3
}
```

What's expected:
(I've tried it on https://hocon-playground.herokuapp.com/ )
```
a:aa: "1"
a:bb: "2"
a:cc: "3.3"
```

But was:
```
a:cc: "3.3"
```

**Comment by Arkatufus** [comment]
Working on a fix, thank you for reporting.

**Comment by iron9light** [comment]
I've fixed this bug. See my PR #91

**Comment by iron9light** [comment]
Another PR #88 . Please check it out.

**Comment by Arkatufus** [comment]
Its OK, I've made a fix for both #89 and #92, they're related.
